### PR TITLE
Fixed incorrect assumption in wfs=false test

### PR DESCRIPTION
**Pull request by mchacki**
### Scope & Purpose

*Test only bugfix. Guarding code against illegal type access.

AddOn: Jenkins detected a real issue within this test.
In the specific test code there was a race, where the test expects that we cannot elect a new leader,
but in fact we can.
If we have 3 servers A(leader),B,C
And we stop one of them (say B)
Than A and C can continue.
If then C is stopped.
A is alone, and cannot continue. (until failovers happen, but that is not in this test)
If now B is rebooted, B can actually vote in same cases, and therefore A and B can continue. 
(This is done via a security message of the log) as A and C both are guaranteed to have all confirmed data at least in memory.
However we stop and continue only C (B still stopped) A and C can for simplicity reasons right now, not continue until B shows up, to avoid wfs=false dataloss scenario.
*

- [X] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.11: *(Please link PR)*
  - [ ] Backport for 3.10: *(Please link PR)*
  - [ ] Backport for 3.9: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

**Review by jsteemann — APPROVED** [review]
LGTM

**Review by hkernbach — APPROVED** [review]

**Review by neunhoef — APPROVED** [review]
LGTM

File: tests/js/server/replication2/replication2-supervision-wait-for-sync-false.js [hunk 1]
```diff
@@ -114,10 +129,23 @@ const replicatedLogWaitForSyncFalseSupervisionSuite = function () {
...
-      assertEqual(status.local.spearhead.index, logIndex);
+      // There is actually a race on the indexes in the log here, that has the same reason
+      // why we need to be careful with stop / continue ordering of servers.
+      // The leader will try to send a message as soon as it get's aware of a stopped server.
+      // That is after the server is stopped, and agency times out on the servers heartbeat.
+      // This message can be either before the expected logindex or after.
+      // As we are shutting down 2 servers, we will have up to 2 messages in this race.
+      // (Where the message of the first one only is highly unlikely)
+
+      // We can be up to 2 entries further ten the expected logIndex
```
Comment (jsteemann) [inline tests/js/server/replication2/replication2-supervision-wait-for-sync-false.js:143]:
```suggestion
      // We can be up to 2 entries further than the expected logIndex
```

File: tests/js/server/replication2/replication2-supervision-wait-for-sync-false.js [hunk 2]
```diff
@@ -114,10 +129,23 @@ const replicatedLogWaitForSyncFalseSupervisionSuite = function () {
...
           }
           return true;
-        },
+        }
       );
       const {participants: {[leader]: {response: status}}} = log.status();
-      assertEqual(status.local.spearhead.index, logIndex);
+      // There is actually a race on the indexes in the log here, that has the same reason
+      // why we need to be careful with stop / continue ordering of servers.
+      // The leader will try to send a message as soon as it get's aware of a stopped server.
```
Comment (jsteemann) [inline tests/js/server/replication2/replication2-supervision-wait-for-sync-false.js:137]:
```suggestion
      // The leader will try to send a message as soon as it gets aware of a stopped server.
```

**Review by jsteemann — APPROVED** [review]
LGTM

**Comment by mchacki** [comment]
Thank you for fixing my typos :+1:

**Pull request merged**